### DISCUSSION TITLE
Can't retrieve git-tags on bitbucket.org

### DISCUSSION THREAD

**Discussion by andrefmarques** [renovatebot/renovate#30115] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Self hosted on kubernetes, using the helm chart v37.425.1

### Please tell us more about your question or problem

Hi! I have renovate deployed with the helm chart (v37.425.1) configured to access a repository with terraform modules and instantiations of the modules.
This is the config:
```
{
  "platform": "bitbucket",
  "repositories": [
    "organization/repository"
  ]
}
```

And I also have this environment variables:
```
RENOVATE_USERNAME=<username>
RENOVATE_PASSWORD=<password>
```

With this setup I'm able to access the repository, but I'm unable to retrieve git tags.
On the repository I have this configuration file:
renovate.json:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "ignorePaths": ["**/examples/**"],
}
```

And an example of an instantiation of a module defined on the same repository:
```
module "instantiation" {
  source = "git@bitbucket.org:organization/repository.git//module?ref=module/1.0.0"
  [...]
}
```

With this setup renovate can't retrieve the git tags:
```
DEBUG: Datasource unknown error (repository=organization/repository)
       "datasource": "git-tags",
       "packageName": "git@bitbucket.org:organization/repository.git//module",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "git@bitbucket.org:organization/repository.git//module"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n",
         "stack": "Error: Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n\n    at Object.action (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:54:29)\n    at /usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:121:14)\n    at GitExecutorChain.<anonymous> (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:97:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/dist/cjs/index.js:52:24)"
       }
DEBUG: Failed to look up git-tags package git@bitbucket.org:organization/repository.git//module (repository=organization/repository, packageFile=path/file.tf, dependency=git@bitbucket.org:organization/repository.git//module)
```

I've configured the username and password on the repository, using the same credentials used to parse the repository:
renovate.json:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "ignorePaths": ["**/examples/**"],
  "hostRules": [
    {
      "matchHost": "bitbucket.org",
      "username": "process.env.RENOVATEBOT_USERNAME",
      "password": "process.env.RENOVATEBOT_PASSWORD"
    }
  ]
}
```

However it still fails, this time with Authentication failed:
```
DEBUG: Found hostRules username and password for url git@bitbucket.org:organization/repository.git//module (repository=organization/repository)
DEBUG: Datasource unknown error (repository=organization/repository)
       "datasource": "git-tags",
       "packageName": "git@bitbucket.org:organization/repository.git//module",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "https://**redacted**@bitbucket.org/organization/repository.git//module"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "remote: Invalid credentials\nfatal: Authentication failed for 'https://bitbucket.org/organization/repository.git//module/'\n",
         "stack": "Error: remote: Invalid credentials\nfatal: Authentication failed for 'https://**redacted**@3.25.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:54:29)\n    at /usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:121:14)\n    at GitExecutorChain.<anonymous> (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:97:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/renovate/node_modules/.pnpm/simple-git@3.25.0/node_modules/simple-git/dist/cjs/index.js:52:24)"
       }
DEBUG: Failed to look up git-tags package git@bitbucket.org:organization/repository.git//module (repository=organization/repository, packageFile=path/file.tf, dependency=git@bitbucket.org:organization/repository.git//module)
```

I've tried to replicate the command above locally using my personal account, and I get the same error:
             
git ls-remote https://bitbucket.org/organization/repository.git//module

But it works if I use ssh instead of https:
git ls-remote git@bitbucket.org:organization/repository.git//module

In short, renovate can parse the repository but can't get the git tags in the same repo.
Is it possible to configure the git-tags datasource to use the method used to parse the repository?
Also, there is a bitbucket-tags datasource, but the tags here are recognized as git-tags, is this an issue?

**Comment by andrefmarques** [comment, marked as answer]
Solved, changed module instantiation from
```
module "instantiation" {
  source = "git@bitbucket.org:organization/repository.git//module?ref=module/1.0.0"
  [...]
}
```
to
```
module "instantiation" {
  source = "git@bitbucket.org/organization/repository.git//module?ref=module/1.0.0"
  [...]
}
```
